_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,356 @@
+
+```html
+<table>
+	<thead>
+		<tr>
+			<th>Projects</th>
+			<th>Exams</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td colspan="2">15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 4
+
+Each row and column header element has assigned cells, within the same element having a [semantic role][] of `grid`.
+
+```html
+<table role="grid">
+	<thead>
+		<tr role="row">
+			<td></td>
+			<th role="columnheader">Breakfast</th>
+			<th role="columnheader">Lunch</th>
+			<th role="columnheader">Dinner</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr role="row">
+			<th scope="row" role="rowheader">Day 1</th>
+			<td>8:00</td>
+			<td>13:00</td>
+			<td>18:00</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Passed Example 5
+
+Each column header is assigned to a cell. Usage of `headers` attribute changes the relationship between column headers and cells.
+
+```html
+<table>
+	<tr>
+		<th id="col1">Column 1</th>
+		<th id="col2">Column 2</th>
+	</tr>
+	<tr>
+		<td></td>
+	</tr>
+	<tr>
+		<td headers="col2"></td>
+	</tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The column header ("Column 2"), does not have an assigned cell within the same `table` element.
+
+```html
+<table>
+	<thead>
+		<tr>
+			<th>Column 1</th>
+			<th>Column 2</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<td>15%</td>
+		</tr>
+	</tbody>
+</table>
+```
+
+#### Failed Example 2
+
+The column header ('Column2') does not have an assigned cell within the same `table` element. In this case the usage of `headers` attribute removes cell association to the column.
+
+```html
+<table>
+	<tr>
+		<th id="col1">Column 1</th>
+		<th id="Column2">Column 2</th>
+	</tr>
+	<tr>
+		<td></td>
+		<td headers="col1"></td>
+	</tr>
+</table>
+```
+
+#### Failed Example 3
+
+The column header ('Col B') does not have an assigned cell within the same `table` element.
+
+```html
+<div role="grid">
+	<div role="row">
+		<div role="columnheader">Col A</div>
+		<div role="columnheader">Col B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">1A</div>
+		<div role="gridcell">1B</div>
+	</div>
+	<div role="row">
+		<div role="gridcell">2A</div>
+	</div>
+</div>
+```
+
+### Inapplicable
+
+#### Inapplicable Example 1
+
+There is no element with a [semantic role][] of header within the table.
+
+```html
+<table>
+	<tr>
+		<td>12:00</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 2
+
+There are no element with a [semantic role][] of header.
+
+```html
+<table></table>
+```
+
+#### Inapplicable Example 3
+
+There are no elements with a [role][semantic role] of header within the table.
+
+```html
+<table>
+	<tr>
+		<th role="cell">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
+```
+
+#### Inapplicable Example 4
+
+The only element with a [role][semantic role] of header is not [visible][].
+
+```html
+<table>
+	<tr>
+		<th style="display: none;">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
+```
+
```

Comment:
display:none not only makes the element non-visible, it also removes it from the accessibility tree, I believe.